### ISSUE TITLE
Problem with SDM630  Error: Invalid slave ID 

### ISSUE THREAD

**Issue by MarioST2023** [nygma2004/sdm1202mqtt#3]
Hello, 
the Code works fine with SDM72, but at SDM 630 is any Problem. 
The Slave ID is in Code and in SDM630 "2" but it got som Error: Invalid slave ID

Can you help me?

WbR 
Mario

**Comment by nygma2004** [comment]
SDM630 is 3 phase, so you need to read a lot more registers to get all the data out.

If you get an error that it is an invalid slave ID, than your SDM630 is configured to a different slave ID. Adjust the SDM or the code.

**Comment by MarioST2023** [comment]
The Slave Id is the same!  SDM 72 is also 3 Phase and it works without Problem. Is it possible that i have to check an other Setting?

**Comment by nygma2004** [comment]
Yes, please check all the other communication setting. By default the code is set to 9600 baud, which you can change in the settings, but the rest (8 data bits, 1 stop bit, no parity) is set in the code.

**Comment by dg4lax** [comment]
Is a version for the SDM630 planned? Your version for the SDM120 runs fantastic. I would like to read my SDM630 and send the data to my MQTT server. I'm not a professional when it comes to programming. All I need is to copy a few lines of code together and change small things. An online search has so far been unsuccessful.

Greetings Torsten

**Comment by nygma2004** [comment]
I don't have a SDM630, I modified the code completely blind, but based on the documentation, it should be correct.
[sdm6302mqtt.zip](https://github.com/nygma2004/sdm1202mqtt/files/14584880/sdm6302mqtt.zip)
Please give it a try and let me know if this works. I only made changes to the MQTT sending, the webpage does not show the SDM630 values yet. I may also have missed a few registers along the way.

**Comment by dg4lax** [comment]
Thank you for your efforts. I'll try this out tomorrow. I'll tackle the web interface to display the values myself. I think it won't be that difficult. It was important to me to integrate mqtt in order to integrate the values into my FHem (https://wiki.fhem.de/wiki/Hauptseite) installation. I'll let you know if it works.

Greetings Torsten

translated by google :-)

**Comment by dg4lax** [comment]
Hello,
your sketch works fine. I'm rebuilding the web interface myself. This way I can keep myself busy with programming.
I wish you a nice weekend and thank you for the help.

Greetings Torsten

**Comment by nygma2004** [comment]
Hi Torsten,

Are you getting all the data that you need? Because I can see a few more registers on total energy, and I was not sure which is the one you wanted.
I am glad that it is working now. I think we should publish the code for others to be used as well. If you get stuck with the UI, let me know.

**Comment by dg4lax** [comment]
Hello,

Some of the registers are still missing, that's true. Yesterday evening I tried to find out how I can extend your script to include these registers. But I'm not quite there yet :-)
When I look at the register names in the description of the SDM630 I can't find them anywhere in the scetch :-)
I have to really read into it.

But yesterday I found another sketch that I got to run. It outputs all values of the SDM630. The only thing missing is a web interface. But that's not so bad since I integrate the data into my home automation anyway. I have attached the sketch for you as an attachment. Also two screenshots of how I then integrate the data into my FHem installation. The SDM630 data now comes from the other sketch I found and the SDM120 data comes from your sketch. Attached is also a screenshot of the MQTT broker showing how the data arrives there.

As mentioned above, I am a real beginner when it comes to Arduino. I can still make small, simple changes but then it stops. That's why I'm always grateful when someone takes a look at it.

Greetings Torsten￼￼￼

**Comment by dg4lax** [comment]
Attached are the screenshots and the sketch
[Uploading SDM630Gateway-main.zip…]()
<img width="635" alt="Bildschirmfoto 2024-03-15 um 18 02 47" src="https://github.com/nygma2004/sdm1202mqtt/assets/36261226/bba26b9c-304c-4bbc-9e07-a367986fc0f2">
<img width="992" alt="Bildschirmfoto 2024-03-15 um 18 03 16" src="https://github.com/nygma2004/sdm1202mqtt/assets/36261226/2fb5da82-2f5e-4f46-9aa6-8aa3c74376c1">
<img width="819" alt="Bildschirmfoto 2024-03-15 um 18 14 34" src="https://github.com/nygma2004/sdm1202mqtt/assets/36261226/7e4305cb-a25b-4620-bfc6-e0f2c147c722">

**Comment by nygma2004** [comment]
Something is not right on these screenshots. Where are the voltage1, voltage2, voltage3... extra fields for the 3 phases?

**Comment by dg4lax** [comment]
> Am 19.03.2024 um 18:30 schrieb Csongor Varga ***@***.***>:
> 
> Something is not right on these screenshots. Where are the voltage1, voltage2, voltage3... extra fields for the 3 phases?
The SDM630 is currently in my office and is only connected to one phase :-) The data from the other 2 phases also work.

**Comment by nygma2004** [comment]
Please send me new pic when all 3 phases are connected and sending data :) And also the webpage if you modified it.

**Comment by dg4lax** [comment]
This is the reading when all 3 phases are connected. But it's always the same phase :-)

<img width="630" alt="Bildschirmfoto 2024-03-19 um 19 44 23" src="https://github.com/nygma2004/sdm1202mqtt/assets/36261226/3ec3ea4b-9441-4cf2-bc60-3c533ebab8db">
<img width="295" alt="Bildschirmfoto 2024-03-19 um 19 44 51" src="https://github.com/nygma2004/sdm1202mqtt/assets/36261226/50de815d-1935-4f0f-b818-b789afa742a7">
